politeiawww/email.go

```diff
@@ -162,18 +163,86 @@ func (p *politeiawww) emailAuthorForVettedProposal(proposal *www.ProposalRecord,
 		StatusChangeReason: proposal.StatusChangeMessage,
 	}
 
-	subject := "Your Proposal Has Been Published"
-	body, err := createBody(templateProposalVettedForAuthor, &tplData)
+	subject := "Your Proposal Has Been Censored"
+	body, err := createBody(templateProposalCensoredForAuthor, &tplData)
 	if err != nil {
 		return err
 	}
 
 	return p.sendEmailTo(subject, body, authorUser.Email)
 }
 
-// emailAuthorForCensoredProposal sends an email notification for a new
-// proposal becoming censored to the proposal's author.
-func (p *politeiawww) emailAuthorForCensoredProposal(proposal *www.ProposalRecord, authorUser *user.User, adminUser *user.User) error {
+// emailProposalStatusChange sends emails regarding the proposal status change
+// event. Sends email for the author and the users with this notification
+// bit turned on. This function replaces tree functions from the old event
+// manager: emailAuthorForVettedProposal, emailUsersForVettedProposal,
+// emailAuthorForCensoredProposal
+func (p *politeiawww) emailProposalStatusChange(data dataProposalStatusChange, emails []string) error {
+	if p.smtp.disabled {
+		return nil
+	}
+
+	route := strings.Replace(guiRouteProposalDetails, "{token}", data.token, 1)
+	l, err := url.Parse(p.cfg.WebServerAddress + route)
+	if err != nil {
+		return err
+	}
+
+	var subject string
+	var template *template.Template
+
+	// Prepare and send author's email if author notification bit
+	// is set
+	if notificationIsSet(data.emailNotifications,
+		www.NotificationEmailMyProposalStatusChange) {
+		switch data.status {
+		case www.PropStatusCensored:
+			subject = "Your Proposal Has Been Censored"
+			template = templateProposalCensoredForAuthor
+		case www.PropStatusPublic:
+			subject = "Your Proposal Has Been Published"
+			template = templateProposalVettedForAuthor
+		}
+		authorTplData := proposalStatusChangeTemplateData{
+			Link:               l.String(),
+			Name:               data.name,
+			StatusChangeReason: data.statusChangeMessage,
+		}
+		body, err := createBody(template, &authorTplData)
+		if err != nil {
+			return err
+		}
+		err = p.sendEmailTo(subject, body, data.email)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Prepare and send user's email, if there is any
+	if len(emails) > 0 {
```

Comment:
Instead of doing this here and having the send email code nested, I would just check this at the top of the function and return immediately if there are no emails.